rust/lance/src/dataset/scanner.rs

```diff
@@ -506,10 +506,7 @@ impl Scanner {
                 message: format!("Column {} not found", q.column),
                 location: location!(),
             })?;
-            let vector_field = ArrowField::try_from(vector_field).map_err(|e| Error::IO {
-                message: format!("Failed to convert vector field: {}", e),
-                location: location!(),
-            })?;
+            let vector_field = ArrowField::from(vector_field);
```

Comment:
This isn't quite the same is it?  The old style would return an error and this new style panics?

Reply:
The conversion from Field to Arrow::Field is safe

https://github.com/lancedb/lance/blob/main/rust/lance-core/src/datatypes/field.rs#L608